assets/src/editor/events/dates-and-times/editor-date/edit-form/use-date-entity-input-config.js

```diff
@@ -131,11 +133,13 @@ const useDateEntityInputConfig = ( {
 			label: __( 'Capacity', 'event_espresso' ),
 			default: -1,
 			changeListener: ( value ) => {
-				dateEntity.regLimit = parseInfinity(
+				const regLimit = parseInfinity(
 					value || -1,
 					true,
 					true
 				);
+				dateEntity.regLimit = regLimit;
```

Comment:
Although this is not directly related to this PR.
We need to reconsider how state management is done.
One way of improvement is by using immutability.

Reply:
It's already immutable :)

Reply:
> It's already immutable

???

The model entities are not immutable, some fields that use value objects like date or money fields are immutable, but the objects in whole are not.

or are you referring to the state object itself? In which case, yes the state is immutable.

> We need to reconsider how state management is done.
> One way of improvement might be: immutability.

How would that work using the code above as an example? 
Are you suggesting constructing a whole new event date object when the capacity (regLimit) property changes value?

Reply:
`Are you suggesting constructing a whole new event date object when the capacity (regLimit) property changes value?`
Yes, that would be faster since old object will be garbage collected and memory got freed.
From what I saw we already use that pattern since we are using `@wordpress/data`